model/src/pyrenew/regression.py

```diff
@@ -114,38 +138,54 @@ def predict(
         coefficients : ArrayLike
             Sampled prediction coefficients distribution generated
             from coefficients priors.
+        predictor_values : ArrayLike(n_predictors, n_observations)
+            Matrix of predictor variables (covariates) for the
+            regression problem. Each row should represent the
+            predictor values corresponding to an observation;
+            each column should represent a predictor variable.
+            You do not include values of 1 for the intercept;
+            these will be added automatically.
 
         Returns
         -------
         ArrayLike
             Array of transformed predictions.
         """
-        transformed_prediction = (
-            intercept + self.fixed_predictor_values @ coefficients
-        )
+        transformed_prediction = intercept + predictor_values @ coefficients
         return self.transform.inv(transformed_prediction)
 
-    def sample(self) -> dict:
+    def sample(self, predictor_values: ArrayLike) -> dict:
         """
         Sample generalized linear model
 
+        Parameters
+        -----------
+        predictor_values : ArrayLike(n_predictors, n_observations)
+            Matrix of predictor variables (covariates) for the
+            regression problem. Each row should represent the
+            predictor values corresponding to an observation;
+            each column should represent a predictor variable.
+            Do not include values of 1 for the intercept;
+            these will be added automatically. Passed as the
+            `predictor_values` argument to
+            :meth:`GLMPrediction.predict()`
+
         Returns
         -------
-        dict
-            A dictionary containing transformed predictions, and
-            the intercept and coefficients sample distributions.
+        GLMPredictionSample
         """
         intercept = numpyro.sample(
             self.name + self.intercept_suffix, self.intercept_prior
         )
         coefficients = numpyro.sample(
             self.name + self.coefficient_suffix, self.coefficient_priors
         )
-        prediction = self.predict(intercept, coefficients)
-        return dict(
-            prediction=prediction,
-            intercept=intercept,
-            coefficients=coefficients,
+        prediction = self.predict(intercept, coefficients, predictor_values)
+
+        return GLMPredictionSample(
+            prediction=SampledValue(prediction),
+            intercept=SampledValue(intercept),
+            coefficients=SampledValue(coefficients),
         )
 
     def __call__(self, **kwargs):
```

Comment:
```suggestion
    def __call__(self, *args, **kwargs):
```

Reply:
And similarly for the call to `sample()` within

Reply:
included in cf3b96a7f790b891ec1df0141542bcb9863dd365